linker/Mono.Linker.Steps/SweepStep.cs

```diff
@@ -336,13 +355,79 @@ void SweepDebugInfo (Collection<MethodDefinition> methods)
 			}
 		}
 
-		protected void SweepCollection<T> (IList<T> list) where T : IMetadataTokenProvider
+		protected void SweepCollection<T> (IList<T> list, AssemblyAction action) where T : IMetadataTokenProvider
 		{
 			for (int i = 0; i < list.Count; i++)
-				if (!Annotations.IsMarked (list [i])) {
-					ElementRemoved (list [i]);
-					list.RemoveAt (i--);
+				if (!Annotations.IsMarked (list[i])) {
+					if (action == AssemblyAction.AddBypassNGen) {
+						MethodDefinition method = list[i] as MethodDefinition;
+						if (method != null) {
+							method.CustomAttributes.Add(bypassNgenAttribute);
+						}
+					}
+					else {
+						ElementRemoved(list[i]);
+						list.RemoveAt(i--);
+					}
+				}
+		}
+
+		private CustomAttribute GetBypassNGenAttribute (ModuleDefinition targetModule)
+		{
+			AssemblyDefinition coreLibAssembly = Context.Resolve (targetModule.TypeSystem.CoreLibrary);
+			ModuleDefinition corelibMainModule = coreLibAssembly.MainModule;
+			TypeReference bypassNGenAttributeRef = new TypeReference ("System.Runtime", "BypassNGenAttribute", corelibMainModule, targetModule.TypeSystem.CoreLibrary);
+			TypeDefinition bypassNGenAttribute = corelibMainModule.MetadataResolver.Resolve (bypassNGenAttributeRef);
+			MethodDefinition bypassNGenAttributeDefaultConstructor = null;
+
+			if (bypassNGenAttribute == null) {
+				// System.Runtime.BypassNGenAttribute is not found in corelib. Add it.
+				TypeReference systemAttributeRef = new TypeReference ("System", "Attribute", corelibMainModule, targetModule.TypeSystem.CoreLibrary);
+				TypeReference systemAttribute = corelibMainModule.MetadataResolver.Resolve (systemAttributeRef);
+				systemAttribute = corelibMainModule.ImportReference (systemAttribute);
+
+				if (systemAttribute == null)
+					throw new System.ApplicationException ("System.Attribute is not found in " + targetModule.TypeSystem.CoreLibrary.Name);
+
+				MethodReference systemAttributeDefaultConstructorRef = new MethodReference (".ctor", corelibMainModule.TypeSystem.Void, systemAttributeRef);
+				MethodReference systemAttributeDefaultConstructor = corelibMainModule.MetadataResolver.Resolve (systemAttributeDefaultConstructorRef);
+				systemAttributeDefaultConstructor = corelibMainModule.ImportReference (systemAttributeDefaultConstructor);
+
+				if (systemAttributeDefaultConstructor == null)
+					throw new System.ApplicationException ("System.Attribute has no default constructor");
+
+				bypassNGenAttribute = new TypeDefinition (
+										"System.Runtime",
+										"BypassNGenAttribute",
+										TypeAttributes.NotPublic | TypeAttributes.Sealed,
+										systemAttribute);
+
```

Comment:
Strange formatting - maybe put the parameters just one indentation level past the previous line?

Reply:
I fixed formatting by just moving the args to the call line.